spec/models/menu_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe Menu, type: :model do
+  describe 'associations' do
+    it { is_expected.to belong_to(:user).with_foreign_key('created_by') }
+  end
+
+  describe '#products' do
+    let(:product_1) { create(:product, name: 'Product 1') }
```

Comment:
this case, you haven't use product name, so i think you shouldn't define name for a product.